src/main/scala/ai/datahunters/md/reader/SolrHashReader.scala

```diff
@@ -0,0 +1,81 @@
+package ai.datahunters.md.reader
+
+import java.util
+
+import ai.datahunters.md.config.writer.SolrWriterConfig
+import ai.datahunters.md.util.SolrClientBuilder
+import org.apache.solr.client.solrj.impl.CloudSolrClient
+import org.apache.solr.common.SolrDocument
+import org.apache.solr.common.params.MapSolrParams
+import org.apache.spark.sql.types.{DataTypes, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+
+import scala.collection.JavaConversions._
+
+/**
+  * Read  id and hashes to DataFrame mainly for processing reduction
+  *
+  * @param sparkSession
+  * @param config
+  */
+case class SolrHashReader(sparkSession: SparkSession,
+                          config: SolrWriterConfig) extends PipelineSource {
+
+  import ai.datahunters.md.reader.SolrHashReader._
+
+  private val documentToStrings: SolrDocument => Seq[String] =
+    (document: SolrDocument) => HashSeq.map(s => getDocumentFieldValueOrEmptyString(document, s))
+
+
+  override def load(): DataFrame = {
+    val client = SolrClientBuilder().setZKServers(config.zkServers)
+      .setZKSolrChroot(config.zkSolrZNode)
+      .setDefaultCollection(config.collection)
+      .build()
+
+    val documents = getSolrDocumentList(client)
+
+    val idHashValueSeq = documents.map((d:SolrDocument) =>
+      Row.fromSeq(documentToStrings(d)))
+
+    val dataRdd = sparkSession.sparkContext.parallelize(idHashValueSeq)
+    sparkSession.createDataFrame(dataRdd, Schema)
+  }
+
+  private def getDocumentFieldValueOrEmptyString(document: SolrDocument, fieldName: String) = {
+    val fieldValue = Option(document.getFirstValue(fieldName).asInstanceOf[String])
+    fieldValue.orNull
+  }
+
+  def getSolrDocumentList(client: CloudSolrClient) = {
+    val queryParamMap = new util.HashMap[String, String]()
+    queryParamMap.put(QueryKey, QueryAllValue)
+    queryParamMap.put(FieldKey, HashSeq.mkString(Comma))
+    val queryParams = new MapSolrParams(queryParamMap)
+    val response = client.query(queryParams)
+    response.getResults
+  }
+}
+
+object SolrHashReader {
+  val QueryAllValue = "*:*"
+  val Comma = ","
+  val QueryKey = "q"
+  val FieldKey = "fl"
+
+  val Crc32 = "hash_crc32"
```

Comment:
@dawidsau I'm pretty sure we have those consts, maybe not exactly the same - without prefix but it would be better to use them. Second thing I'm afraid of is naming convention. We use snake case for our Solr schema but it's not always the case. I think we should get naming convention from props (existing one) and use converting method which will prepare final names. To achieve that, original names should contain space.

Reply:
Done.